code/global/RPCs.cs

```diff
@@ -113,6 +113,12 @@ public static void ClientConfirmPlayer(TTTPlayer confirmPlayer, PlayerCorpse pla
             }
         }
 
+        [ClientRpc]
+        public static void ClientOnMovedToForceSpectator(bool value)
+        {
+            Hud.Current.AliveHudInstance.Enabled = false;
+        }
```

Comment:
We need this? I feel like this should be done inside of `MakeSpectator`

Reply:
MakeSpectator does set its health and state, but is not what calls "OnKilled", "OnKilled" is what calls MakeSpectator. Also, we cannot manually call this method, which will cause a stack overflow. Do you have a way for me to trigger OnKilled without causing this?

https://github.com/TTTReborn/tttreborn/blob/24eac8ada69f3cf43b96f5688767e9f8ecc76688/code/player/TTTPlayer.Spectating.cs#L74-L85

https://github.com/TTTReborn/tttreborn/blob/24eac8ada69f3cf43b96f5688767e9f8ecc76688/code/rounds/InProgressRound.cs#L24-L37

Reply:
Hmm... I'm thinking maybe `MakeSpectator` is what calls for the disable/deletion of HUD. There is definitely some way to handle this.

Reply:
I agree, but don't I have to do this on client side?

Reply:
here are some options, you'd have to give me some time to go digging around for the best one.

1. `MakeSpectator` damages the player for a huge amount, thus killing the player and triggering `OnKill` (which then disables the UI)
2. `MakeSpectator` runs the `ttt_player_onkilled` event which triggers the RPC to disable the hud

Reply:
I manually triggered OnKill inside of ToggleForcedSpectator, if this were to be written we would have a stack overflow. Seems that we still have the issue with AliveHud even after @Alf21's changes.